src/oscar/apps/offer/benefits.py

```diff
@@ -203,6 +204,50 @@ def apply(
         return BasketDiscount(discount)
 
 
+class AbsoluteProductDiscountBenefit(AbsoluteDiscountBenefit):
+    """
+    An offer benefit that gives an absolute discount on each applicable product.
+    The value for this benefit refers to the maximum discount per product
+    regardless of the quantity.
+    """
+
+    class Meta:
+        app_label = "offer"
+        proxy = True
+        verbose_name = _("Absolute product discount benefit")
+        verbose_name_plural = _("Absolute product discount benefits")
+
+    def get_lines_to_discount(self, offer, line_tuples):
+        # Determine which lines can have the discount applied to them
+        max_affected_items = self._effective_max_affected_items()
+        num_affected_items = 0
+        lines_to_discount = []
+        for price, line in line_tuples:
+            if num_affected_items >= max_affected_items:
+                break
+            qty = min(
+                line.quantity_without_offer_discount(offer),
+                max_affected_items - num_affected_items,
+            )
+            lines_to_discount.append((line, price, qty))
+            num_affected_items += qty
+        return lines_to_discount
+
+    def apply(self, basket, condition, offer, **kwargs):
+        # Fetch basket lines that are in the range and available to be used in an offer.
+        line_tuples = self.get_applicable_lines(offer, basket)
+        lines_to_discount = self.get_lines_to_discount(offer, line_tuples)
+
+        applied_discount = D("0.00")
+        for (line, price, qty) in lines_to_discount:
+            # If price is less than the fixed discount, then it will be free.
+            line_discount = min(price * qty, self.value)
```

Comment:
Shouldn't self.value also be multiplied with qty?

Reply:
If we multiply self.value with qty then user can order unlimited products for free. Using the current configuration the maximum discount a user can avail per product would be self.value regardless of the quantity.

Reply:
I don't understand, this value is the maximum discount for the LINE right, not per product.
But right now, if there are 2 products and the discount value is 1, I will only get discount 1 and not 2, which is 1 for each product.

Reply:
I also don't think that it is a problem that unlimited ordering products for free is a problem. Each benefot has a max applications property the use can use to limit that, and also if the user does not choose a discount value greater than the value of the product there is no free product. With the percentage discount you can also get unlimited products for free if you put the percentage to 100. Making sure the product isn't free is the responsibility of the user.